### PR TITLE
David

### PR DESCRIPTION
**Pull request by whymand**
Fixed hard-coding issues with labelling and column renaming

### CODE REVIEW

File: univ_rd/univ_rd.R [hunk 1]
```diff
@@ -36,12 +36,14 @@ county_univRD <- NSF_univRD %>%
...
 
-labels<-c("total R&D spending",
-          "total deflated business financed R&D spending",
-          "FIPS code")
+labels<-c(rd_total_biz = "total deflated business financed R&D spending",
+          rd_total =  "total R&D spending",
+          stco_code = "FIPS code",
+          cbsa_code = "FIPS code")
+
+label(county_univRD) = as.list(labels[match(names(county_univRD), names(labels))])
```
Comment (whymand) [inline univ_rd/univ_rd.R]: This line of code assigns the "labels" vector to the dataframe in a soft-coded manner. The vector's elements are paired (column name = label) and the "match" feature ensures that the label is assigned to the corresponding column name in the dataset: *even if the columns in the dataset are in a different order or if some columns are absent, no mislabeling will occur.

File: export_monitor/export_monitor.R [hunk 2]
```diff
@@ -17,46 +17,44 @@ if (any(!check)) {
...
-  "cbsa_gdp_nominal",
-  "cbsa_gdp_real",
-  "cbsa_2003_2014_pct_gdp_growth",
-  "cbsa_2008_2017_pct_gdp_growth",
-  "cbsa_2003_2017_pct_gdp_growth",
-  "cbsa_2014_2017_pct_gdp_growth",
-  "cbsa_pct_gdp_growth",
-  "cbsa_jobs")
+
+cbsa_export<-select(cbsa_export,
```
Comment (whymand) [inline export_monitor/export_monitor.R]: variable renaming using the select function (avoids hard code issue)

File: export_monitor/export_monitor.R [hunk 3]
```diff
@@ -17,46 +17,44 @@ if (any(!check)) {
 #data up to 2017 exists
 
 cbsa_export <- readxl::read_xlsx("V:/Export Monitor/2018/Deliverables/Deliverables/Metros Data/Metros by Total, NAICS 2 3.xlsx", sheet = "Total") %>%
-  filter(Year == 2017) %>%
-  select(-c(2,13:16))
+  filter(Year == 2017)
+
 
 #use old column names as labels (use View() to see labels)
 set_label(cbsa_export)<-colnames(cbsa_export)
```
Comment (whymand) [inline export_monitor/export_monitor.R]: old column names were assigned as labels before any column deletions or renaming occurs. At the point when column renaming and deletions begin (see below), each column has already been labelled, so removing or renaming columns will not 'break' the labelling structure or lead to mislabelling

**Comment by fansi-sifan** [comment]
Hey, unfortunately I can't merge your request because there are conflicting files - it could be that you didn't pull the latest commits from the master branch before making changes. 

See if you can resolve it on your end, but let me know if you think it's going to take too long and I will do it manually on the master. Thanks

**Pull request closed without merging**